### PR TITLE
Add simple Moderation extension - V1

### PR DESCRIPTION
**Pull request by Shadwz17**
I added a simple moderation extension that allows the staff members to ban, kick, unban and purge messages. This is the first version of the moderation extension I'm doing so maybe in the future I'll add some other commands.

Images:
https://imgur.com/022bhNw - Unban - Usage ::unban ID
https://imgur.com/WfhqTSd - Purge - Usage ::purge NUMBER_OF_MESSAGES
https://imgur.com/jM5NFDn - Ban - Usage ::ban (You need to tag the person)

There is no kick image but it's the same as all the ones above.

### CODE REVIEW

File: bot/extensions/moderation/moderation.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    @has_permissions(kick_members=True)
+    async def kick(self, ctx, member: Member, reason="No reason"):
+            guild = ctx.guild
+            embed = Embed(title="Grace Moderation - KICK", description=f"{member.mention} has been kicked.", timestamp=datetime.datetime.utcnow())
+            embed.add_field(name="Reason: ", value=str(reason), inline=False)
+            await ctx.reply(embed=embed)
+            await guild.kick(user=member)
+
+    @command(name='ban', help="Allows a staff member to ban a user based on their behaviour.")
+    @cooldown(1, 5, BucketType.user)
```
Comment (SoyBison) [inline bot/extensions/moderation/moderation.py:25]: Remove cooldown on ban and kick, add mass ban/kick commands in the future.

**Review by SoyBison — CHANGES_REQUESTED** [review]
Love adding this feature. Main use for me would be to allow other cogs to call these commands in the future.

Make the changes we talked about in the lab, and then I'll test locally after work, and we'll send it up!

File: bot/extensions/moderation/moderation.py (hunk 1 above)
Comment (Shadwz17) [reply bot/extensions/moderation/moderation.py]: Solved on the commit below this.

File: bot/extensions/moderation/moderation.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    def __init__(self, bot):
+        self.bot = bot
+
+    @command(name='kick', help="Allows a staff member to kick a user based on their behaviour.")
+    @has_permissions(kick_members=True)
+    async def kick(self, ctx, member: Member, reason="No reason"):
+            guild = ctx.guild
+            embed = Embed(title="Grace Moderation - KICK", description=f"{member.mention} has been kicked.", timestamp=datetime.datetime.utcnow())
+            embed.add_field(name="Reason: ", value=str(reason), inline=False)
+            await ctx.reply(embed=embed)
```
Comment (SoyBison) [inline bot/extensions/moderation/moderation.py:20]: As in Malassi's suggestion, can you also throw the Embed to the mod logs channel? The channel code should already be in the config class.

**Review by SoyBison — CHANGES_REQUESTED** [review]
Sorry to throw another request, but Malassi is right the embeds should also go to the mod logs.

**Comment by SoyBison** [comment]
We added some thoughts on how to implement the Mod Logs @Shadwz17. We think you should A. Add mod-logs to the database (876592591657918514). B. create a function at the bot level for logging an action.

Welcome to scope creep woo!

**Review by PenguinBoi12 — APPROVED** [review]
1. The moderation cog doesn't need to be in it's own package. It could be placed directly in extensions folder. (Don't worry, I'll make the change)
2. Be careful to follow Python's style convention (PEP-0008) and general good code styling. In general, this isn't an issue but don't forget to always choose good descriptive name (`id` was the user id so `user_id` is cleaner)
3. Don't hesitate to space logical part of your code. 
4. Don't forget to remove the unused imports.

**Pull request merged**